source/lib/vagrant-openstack-provider/config_resolver.rb

```diff
@@ -23,15 +23,14 @@ def resolve_flavor(env)
         flavor
       end
 
-      def resolve_image(env)
+      def resolve_image(env, image_name)
```

Comment:
For code style consistency, i would prefer to keep the method signature taking only the `env` parameter for public methods. We could have something like

```ruby
def resolve_image(env)
  @logger.info 'Resolving image'
  resolve_image_internal(env, env[:machine].provider_config.image)
end

def resolve_volume_boot_image(env)
  @logger.info 'Resolving image to create a volume from'
  resolve_image_internal(env, env[:machine].provider_config.volume_boot[:image])
end

private 

def resolve_image_internal(env, image_name)
  return nil if image_name.nil?

  nova = env[:openstack_client].nova
  env[:ui].info(I18n.t('vagrant_openstack.finding_image'))
  images = nova.get_all_images(env)
  image = find_matching(images, image_name)
  fail Errors::NoMatchingImage unless image
  image
end
```